### DISCUSSION TITLE
Is there a way to disable f/t repeat?

### DISCUSSION THREAD

**Discussion by kvietcong** [folke/flash.nvim#19] [Q&A] [answered]
Is there a way to retain the core f/t enhanced behavior but only allow ;/. to repeat searches?

**Comment by folke** [comment]
nope, it's all or nothing with ftFT

**Reply by kvietcong** [reply]
Thanks for the quick reply. Do you foresee adding a toggle for that or is this kinda set in stone? I tried messing around with the `char.lua` plugin file but just adding an extra check for the ;/. keys where you check for repeats doesn't do anything.

**Reply by folke** [reply]
What exactly do you want to achieve?
What do you mean with the core f/t enhanced behavior?

**Reply by kvietcong** [reply]
I was hoping to make ;/, be the only keys to repeat a search. So pressing f after your first hit would trigger a new search rather than repeat the last one. Essentially how the current f/t work but multiline and case insensitive.

**Reply by simonmandlik** [reply]
I second that, that would make it possible to simulate the vanilla f/t behavior

**Reply by tonyjara** [reply]
I personally love the s behavior but not a fan of f, I ended up here trying to disable f/t altogether.

**Reply by dbanks12** [reply]
Same here! I am loving the s S behavior, but the f F t T repeat is too much for me after a decade of heavy use of their default bindings.

**Reply by elderkeltir** [reply]
Is there a way to actually disable this f/t feature for now?
My problem is the following scenario: `f"vf)d` (I want to find " char then select from this position to ')' char and then delete it). After I click f for the second time - it jumps to another " char and the rest of the motion is broken. example line `log_error(format_str("here is some kind of string %d", some_int))`
tho s/S feature is a literal monster, and I want it to remain in my config.
thanks!

**Reply by chuan2984** [reply]
like @elderkeltir mentioned this annoying default repeat feature ruined so many of my well crafted macros. i really like the highlight with inline search, and its always a problem with inline search(you cant see the next occurence). Unfortunately, today is the day i let go of the f/t mode for good, the plugin is still the goat, but would be lovely to allow only highlight and not default repeat for f,F,t,T

**Reply by simonmandlik** [reply, marked as answer]
I'm using this setting to get the desired vanilla-like behavior

```
modes = { char = { char_actions = function() return { [";"] = "next", [","] = "prev" } end } }
```

**Reply by chuan2984** [reply]
@simonmandlik isnt this the same as turning it off?

**Reply by simonmandlik** [reply]
No, this doesn't turn off flash, only makes it behave like vanilla f/t on steroids (multi-line search and highlighting).

I'm using this config for `char`:

```lua
char = {
    char_actions = function() return { [";"] = "next", [","] = "prev" } end,
    search = { wrap = true },
    highlight = {
        backdrop = false,
        groups = {
            match = "FlashCharMatch",
            label = "FlashCharLabel"
        }
    }
},
```

which gets me this (the first `M` is highlighted differently than the rest, only `;` and `,` are used to repeat the search, `f`,`t`,`F` and `T` start new search)

<img width="811" alt="image" src="https://github.com/user-attachments/assets/fe92a4d8-6857-4248-8c4a-298b953521e8" />

**Reply by chuan2984** [reply]
thankyou very much, this is the answer, i got it! I was lazy loading but forgot to set it to trigger on f,F,t,T

**Comment by MrFinchMkV** [comment]
Don't know when the functionality was added, but now it's possible to get the requested behavior by disabling char.

Just the opts config:
```lua
opts = {
    modes = {
      char = {
        enabled = false
      },
    },
  }
```
Full example:
```lua
return {
  "folke/flash.nvim",
  event = "VeryLazy",
  ---@type Flash.Config
  opts = {
    modes = {
      char = {
        enabled = false
      },
    },
  },
  -- stylua: ignore
  keys = {
    { "<leader>a", mode = { "n", "x", "o" }, function() require("flash").jump() end, desc = "Flash" },
    { "<c-a>", mode = { "c" }, function() require("flash").toggle() end, desc = "Toggle Flash Search" },
  },
}
```

**Reply by simonmandlik** [reply]
No, this turns off f/F/t/T from flash entirely.

Please see my [reply](https://github.com/folke/flash.nvim/discussions/19#discussioncomment-11566691) above
